### PR TITLE
feat(output): fluent-bit integrate OpenSearch plugin

### LINKED ISSUE #284: Fluentbit integrated OpenSearch plugin
### Is your feature request related to a problem? Please describe.

Fluent Bit is a graduated sub-project under the Cloud Native Computing Foundation (CNCF) Fluentd project umbrella. Similar to the parent project, Fluent Bit has hundreds of integrations to common tools such as Kafka, Syslog, Loki, as well as to services like Datadog, Splunk, and New Relic. Now with Fluent Bit 1.9, OpenSearch is included as part of the binary package.
But the fluentbit in the fluent-operator does not have an associated crd, for which we need to integrate the OpenSearch plugin.

### Describe the solution you'd like

The specific writing method can be referred to: https://github.com/fluent/fluent-operator/tree/master/apis
In the meantime, we can refer to this article: https://opensearch.org/blog/technical/2022/03/getting-started-with-fluent-bit-and-opensearch/
The profile transcoding in `secret` should look something like this:
```
[OUTPUT]
   Name opensearch
   Match *
   Host 192.168.2.3
   Port 9200
   Index my_index
   Type my_type
```

### Describe alternatives you've considered

Since OpenSearch is similar to Elasticsearch,for the CRD writing of the OpenSearch plugin, we can specifically draw on this document：
https://github.com/fluent/fluent-operator/blob/master/apis/fluentbit/v1alpha2/plugins/output/elasticsearch_types.go

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mangoGoForward**
Signed-off-by: mango <xu.weiKyrie@foxmail.com>

### What this PR does / why we need it:
Fluent-bit integrate OpenSearch plugin

### Which issue(s) this PR fixes:

Fixes #284

### Does this PR introduced a user-facing change?

```release-note
Action Required: Should upgrade fluent-bit image version to 1.9
```

### CODE REVIEW

File: apis/fluentbit/v1alpha2/plugins/output/open_search_types.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+	// When enabled print the elasticsearch API calls to stdout (for diag only)
+	TraceOutput *bool `json:"traceOutput,omitempty"`
+	// When enabled print the elasticsearch API calls to stdout when elasticsearch returns an error
+	TraceError *bool `json:"traceError,omitempty"`
+	// Use current time for index generation instead of message record
+	CurrentTimeIndex *bool `json:"currentTimeIndex,omitempty"`
+	// Prefix keys with this string
+	LogstashPrefixKey string `json:"logstashPrefixKey,omitempty"`
+	// When enabled, mapping types is removed and Type option is ignored. Types are deprecated in APIs in v7.0. This options is for v7.0 or later.
+	SuppressTypeName string `json:"suppressTypeName,omitempty"`
```
Comment (wenchajun) [inline apis/fluentbit/v1alpha2/plugins/output/open_search_types.go:92]: https://docs.fluentbit.io/manual/pipeline/outputs/opensearch
`When enabled, mapping types is removed and Type option is ignored.`
This appears to have only `on` and `off` values, which seems to be possible with bool types?

**Comment by wenchajun** [comment]
The corresponding crd file does not seem to be generated here

File: apis/fluentbit/v1alpha2/plugins/output/open_search_types.go (hunk 1 above)
Comment (mangoGoForward) [reply apis/fluentbit/v1alpha2/plugins/output/open_search_types.go:92]: Thanks, I would to fix now.

Comment (mangoGoForward) [reply apis/fluentbit/v1alpha2/plugins/output/open_search_types.go]: Done.

**Review by wenchajun — APPROVED** [review]

File: apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go [hunk 2]
```diff
@@ -246,8 +254,30 @@ func Test_FluentBitConfig_RenderMainConfig(t *testing.T) {
 		},
 	}
 
+	openSearchOutput := ClusterOutput{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "fluentbit.fluent.io/v1alpha2",
+			Kind:       "Output",
```
Comment (zhu733756) [inline apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go:260]: Kind shoud be ClusterOutput?

File: apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go [hunk 3]
```diff
@@ -246,8 +254,30 @@ func Test_FluentBitConfig_RenderMainConfig(t *testing.T) {
...
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "opensearch_output_0",
+			Namespace: "testnamespace",
```
Comment (zhu733756) [inline apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go:264]: No namespace I think if use cluster CRD.

File: apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go (hunk 2 above)
Comment (mangoGoForward) [reply apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go:260]: Yes, there should be `ClusterOutput`. I noticed that there are many place used `Output`, `Input` or `Filter` in `fluent-bit config`, seems they should be also need add a `Cluster` prefix, and do we need solve also in this PR?

File: apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go (hunk 3 above)
Comment (mangoGoForward) [reply apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go:264]: Got, I will fix. thanks.

File: apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go (hunk 2 above)
Comment (zhu733756) [reply apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go:260]: Yes. Pls help to correct it.

Comment (benjaminhuo) [reply apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go:260]: The kind should always be clusterxxx for fluentbit crds like this https://github.com/fluent/fluent-operator/blob/master/apis/fluentbit/v1alpha2/clusterinput_types.go#L52

But we needn't change the internal structure to clusterxxx like this: https://github.com/fluent/fluent-operator/blob/master/apis/fluentbit/v1alpha2/clusterinput_types.go#L34

Comment (benjaminhuo) [reply apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go:260]: input, output, filter parser CRDs are all changed to clusterxxx in v1.0.0

Comment (mangoGoForward) [reply apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go]: Done.

File: apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go (hunk 3 above)
Comment (mangoGoForward) [reply apis/fluentbit/v1alpha2/clusterfluentbitconfig_types_test.go]: Done.

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
Thank you for your contribution @mangoGoForward !
If this PR requires fluentbit v1.9.0+, we can release a new version of the operator after fluentbit v1.9.4 is release to fix https://github.com/fluent/fluent-bit/issues/5426 @wenchajun

**Pull request merged**